packages/vite/src/node/plugins/asset.ts

```diff
@@ -234,7 +235,23 @@ export function assetPlugin(config: ResolvedConfig): Plugin {
           config.assetsInclude(chunk.moduleIds[0]) &&
           this.getModuleInfo(chunk.moduleIds[0])?.meta['vite:asset']
         ) {
-          delete bundle[file]
+          if (!importedFiles) {
+            importedFiles = new Set()
+            for (const file in bundle) {
+              const chunk = bundle[file]
+              if (chunk.type === 'chunk') {
+                for (const importedFile of chunk.imports) {
+                  importedFiles.add(importedFile)
+                }
+                for (const importedFile of chunk.dynamicImports) {
+                  importedFiles.add(importedFile)
+                }
+              }
+            }
+          }
```

Comment:
Instead of checking every time like https://github.com/vitejs/vite/pull/19775 does, this code constructs the Set if this look up happens. This is because I think that this look up either doesn't happen at all or it happens many times.